gnucash/gnome-utils/dialog-utils.c

```diff
@@ -876,3 +879,187 @@ gnc_get_negative_color (void)
+ *
+ * this procedure does the real work of displaying a sorted list of
+ * available invoice reports
+ ********************************************************************/
+static void
+update_invoice_list (GtkListStore *store)
+{
+    SCM get_rpt_guids = scm_c_eval_string ("gnc:custom-report-invoice-template-guids");
+    SCM template_menu_name = scm_c_eval_string ("gnc:report-template-menu-name/report-guid");
+    SCM reportlist;
+    SCM rpt_guids;
+    int i;
+    GtkTreeIter iter;
+    GtkTreeModel *model = GTK_TREE_MODEL(store);
+
+    gtk_tree_sortable_set_sort_column_id (GTK_TREE_SORTABLE(store),
+                                          COL_INV_NAME, GTK_SORT_ASCENDING);
+
+    reportlist = scm_call_0 (get_rpt_guids);
+    rpt_guids = reportlist;
+
+    gtk_list_store_clear (store);
+
+    if (scm_is_list (rpt_guids))
+    {
+        /* for all the report guids in the list, store them, with a reference,
+             in the gtkliststore */
+        for (i = 0; !scm_is_null (rpt_guids); i++)
+        {
+            gchar *guid_str = scm_to_utf8_string (SCM_CAR(rpt_guids));
+            gchar *name = gnc_scm_to_utf8_string (scm_call_2(template_menu_name,
+                                                  SCM_CAR(rpt_guids), SCM_BOOL_F));
+
+            gtk_list_store_append (store, &iter);
+            gtk_list_store_set (store, &iter,
+                                COL_INV_NAME, name,
+                                COL_INV_GUID, guid_str,
+                                -1);
+            g_free (name);
+            g_free (guid_str);
+
+            rpt_guids = SCM_CDR(rpt_guids);
+        }
+    }
+}
+
+static gboolean
+select_default (GtkWidget *combo, const gchar *default_guid)
+{
+    GtkTreeModel *model = gtk_combo_box_get_model (GTK_COMBO_BOX(combo));
+    GtkTreeIter iter;
+    gboolean found = FALSE;
+    gboolean valid_iter = gtk_tree_model_get_iter_first (model, &iter);
+
+    while (valid_iter)
+    {
+        gchar *guid;
+        gtk_tree_model_get (model, &iter, COL_INV_GUID, &guid, -1);
+
+        if (g_strcmp0 (default_guid, guid) == 0)
+        {
+            gtk_combo_box_set_active_iter (GTK_COMBO_BOX(combo), &iter);
+            g_free (guid);
+            found = TRUE;
+            break;
+        }
+        g_free (guid);
+        valid_iter = gtk_tree_model_iter_next (model, &iter);
+    }
+    return found;
+}
+
+static void
+combo_changed_cb (GtkComboBox *widget, gpointer user_data)
+{
+    GtkTreeIter iter;
+    QofBook *book = gnc_get_current_book();
+
+    if (gtk_combo_box_get_active_iter (widget, &iter))
+    {
+        GtkTreeModel *model = gtk_combo_box_get_model (widget);
+        gchar *name, *guid;
+        gtk_tree_model_get (model, &iter, COL_INV_NAME, &name,
+                                          COL_INV_GUID, &guid, -1);
+
+        if (g_strcmp0 (name, _("Missing Invoice Report")) == 0)
+            gtk_widget_show (user_data);
+        else
+            gtk_widget_hide (user_data);
```

Comment:
you don't use `gtk_widget_set_visible`? I'm usually one for brevity...

Reply:
That maybe factored out on later revisions, on first passes I tend to see what works.